### PR TITLE
🦟 Pinterest data comes HTML encoded, not URL encoded.

### LINKED ISSUE #13881: amp-pinterest embedPin character problem
**I use amp pinterest for amp page for english characters its ok , but if user name chinese or another language display like image below :** 

I use this embed code
`<amp-pinterest`
`width="345"`
`height="678"`
`data-do="embedPin"`
`data-url="https://www.pinterest.com/pin/192740059037785029/">`
`</amp-pinterest>`

And my  `<meta charset="utf-8">` on header.

2018-03-08_21h00_47

### PR DESCRIPTION
**Pull request by cramforce**
Credit to @jridgewell for doing the original debugging.

Fixes #13881

### CODE REVIEW

File: extensions/amp-pinterest/0.1/util.js [hunk 1]
```diff
@@ -39,20 +39,13 @@ function log(queryParams) {
...
-  let decoded, ret;
-  decoded = '';
-  ret = '';
-  try {
-    decoded = decodeURIComponent(str);
-  } catch (e) {}
-  ret = decoded.replace(/</g, '&lt;');
-  ret = ret.replace(/>/g, '&gt;');
-  return ret;
+  return new DOMParser().parseFromString(str, 'text/html').body.textContent;
```
Comment (kristoferbaxter) [inline extensions/amp-pinterest/0.1/util.js:48]: Do we need to handle error conditions?

https://developer.mozilla.org/en-US/docs/Web/API/DOMParser#Error_handling

Comment (cramforce) [reply extensions/amp-pinterest/0.1/util.js:48]: I think that is quite unlikely. We'd see in canary, but this should run on well formed data

Comment (cramforce) [reply extensions/amp-pinterest/0.1/util.js]: But added. Not worth a round-trip :)

File: extensions/amp-pinterest/0.1/util.js [hunk 2]
```diff
@@ -39,20 +39,17 @@ function log(queryParams) {
...
-  let decoded, ret;
-  decoded = '';
-  ret = '';
   try {
-    decoded = decodeURIComponent(str);
-  } catch (e) {}
-  ret = decoded.replace(/</g, '&lt;');
-  ret = ret.replace(/>/g, '&gt;');
-  return ret;
+    return new DOMParser().parseFromString(str, 'text/html').body.textContent;
```
Comment (jridgewell) [inline extensions/amp-pinterest/0.1/util.js:49]: Are we concerned about performance here? Closure has a lighter weight (perf, but larger file size) approach: https://github.com/google/closure-library/blob/8d4f8222eb3729e5aa50c07e82393b15eaf1ab1a/closure/goog/string/string.js#L605-L649

Comment (cramforce) [reply extensions/amp-pinterest/0.1/util.js:49]: No, I think we don't really care. In practice this will be very rare operations.

**Pull request merged**